File: javascript/wordy/wordy.js
export const answer = (string) => {
  let sanitized = sanitize(string);
  let UknowOperationError = new Error("Syntax error");
  console.log(sanitized);
  console.log(eval(sanitized), "<---- eval sanitized");

  if (eval(sanitized) !== undefined) {
    console.log("pasa");
    return eval(sanitized);
  } else {
    console.log("no pasa");
  }
  // return result;
};
export const getAllSigns = (array) => {
  return array.filter((element) => element === "/" || "+" || "-" || "*");
};
export const doOperation = (arraySan) => {
  let result1 = arraySan.split("/");
  let splited = arraySan.split("");
  let result2 = getAllSigns(splited);

  console.log(result1);
  console.log(splited);
};

export const sanitize = (string) => {
  return string
    .replace("What ", "")
    .replace("is ", "")
    .replace("?", "")
    .replace(/by/g, "")
    .split("plus")
    .join("+")
    .split("minus")
    .join("-")
    .split("divided")
    .join("/")
    .split("multiplied")
    .join("*");
};
// export const getOperation = array => {
//   console.log("from getOperation", array[3]);
//   return array[3];
// };
export const getNumbers = (string) => string.replace(/\D/g, "");

export const stringToArr = (string) => string.split(" +");

File: ruby/wordy/wordy.rb
# frozen_string_literal: true

# Class WordProblem
#
class WordProblem
  def initialize(problem)
    @problem = problem
  end

  def answer
    solve = 0
    @problem.scan(/([^-?0-9]+)(-?[0-9]*)/).each do |operator, number|
      int = number.to_i
      operator = operator.strip

      unless operator.match?(/What is|plus|minus|multiplied by|divided by/)
        raise ArgumentError
      end

      solve = int if operator == 'What is'
      solve += int if operator == 'plus'
      solve -= int if operator == 'minus'
      solve *= int if operator == 'multiplied by'
      solve /= int if operator == 'divided by'
    end

    solve
  end
end
